### ISSUE TITLE
[BUG] <UNKNOWN> header parameter error in version 3.34.0

### ISSUE THREAD

**Issue by philipdouglas** [schemathesis/schemathesis#2407]
### Checklist

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

Since updating to 3.34.0 this morning, our schemathesis runs have started failing with this error on every path:

```
________________________________ GET /v2/users _________________________________
Schema Error

Can not generate data for header parameter "<UNKNOWN>"! It should have either `schema` or `content` keywords defined
```

### To Reproduce

🚨 **Mandatory** 🚨: Steps to reproduce the behavior:

1. Run this command: `st run --experimental=openapi-3.1 --header "Authorization:Bearer placeholder" --checks all error.yaml --base-url=https://stoplight.io/mocks/ev-energy/evenergy/319040510/`
(Note: this uses the mock server from our docs platform, because I couldn't reproduce the error without a sufficiently realistic response from the server.)
2. See error

Please include a minimal API schema causing this issue:

```yaml
openapi: 3.1.0
info:
  title: example
  version: '2.0'
paths:
  /users:
    get:
      responses:
        '200':
          content:
            application/json:
              schema:
                type: array
                items: []
          headers:
            Link:
              $ref: '#/components/headers/link'
      operationId: get-users
components:
  headers:
    link:
      schema:
        type: string
```

### Expected behavior
I'm not sure what we're expecting with the new header validation behaviour but, at the very least, I'd expect an error message telling us what header is the problem.

### Environment

```
- OS: Linux
- Python version: 3.11
- Schemathesis version: 3.34.0
- Spec version: 3.11.9
```

**Comment by Stranger6667** [comment]
Hi! 
Thank you for reporting it!

I'll fix it soon

**Comment by Stranger6667** [comment]
The fix is released in `3.34.1`

**Comment by philipdouglas** [comment]
Thanks for the quick work, that's fixed it for us!
